torch/utils/cpp_extension.py

```diff
@@ -740,6 +763,21 @@ def _check_abi(self):
                    'Please set `DISTUTILS_USE_SDK=1` and try again.')
             raise UserWarning(msg)
 
+    def _check_cuda_version(self):
+        if CUDA_HOME:
```

Comment:
I don't think this is a robust check. For example, on my current Arch Linux setup:
```bash
$ echo $CUDA_HOME

$ which nvcc
/opt/cuda/bin/nvcc
rgommers@machine:~/code/numpy.org (master)$ nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
...
Build cuda_11.3.r11.3/compiler.29745058_0
```
There's a more elaborate function `_find_cuda_home` around line 50 in this file.

Reply:
The problem is that one checks the existence of CUDA in the system by locating nvcc, not checking its version. This check looks for the version listed in line 4, for example:

```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2020 NVIDIA Corporation
Built on Mon_Nov_30_19:08:53_PST_2020
Cuda compilation tools, release 11.2, V11.2.67
Build cuda_11.2.r11.2/compiler.29373293_0
```

The regex will parse the 11.2 string in `Cuda compilation tools, release 11.2, V11.2.67`

Reply:
Sure, but my main point is you are raising an exception unless `CUDA_HOME` is set. I think for the Ubuntu et al. default location of CUDA, it's often found by default and the user doesn't have to set the env var. So that'll break, right?

Reply:
The `CUDA_HOME` variable is set by calling `_find_cuda_home`: https://github.com/pytorch/pytorch/blob/0c916c8a4e94c50ae65c2acdfed348cb94bd26de/torch/utils/cpp_extension.py#L163